modules/dcache/src/main/java/org/dcache/pool/classic/IoQueueManager.java

```diff
@@ -451,16 +461,43 @@ public Serializable call() throws NoSuchElementException
                     comparator = comparator.reversed();
                 }
 
+                // Prepare filters
+                Predicate<MoverRequestScheduler.PrioritizedRequest> clientIpFilter;
+                if (clientHost != null && !clientHost.isEmpty()) {
```

Comment:
Oops, sorry, I was thinking about a different part of the code-base.

Your right, it is possible to inject an empty string.

(The syntax is a little out; the correct syntax would be `-client=""`.  The equals is needed and the double-dash actually signifies that everything subsequent to this is arguments, even if they look like options.)

You can use Guava's `com.google.common.base.Strings` class' `isNullOrEmpty` method to make the test a little bit less verbose.